### ISSUE TITLE
Drea Renthal causes crash

### ISSUE THREAD

**Issue by Zipline20** [Sandrem/FlyCasual#1571]
Scurrg bomber is making a range 3 shot here with a focus token. I had a similar crash before, and I restarted. I suspect the crash may have occurred because I restarted the game. I experienced some sort of stuttering of the screen before the crash. I'm playing on a one plus 5t Android phone on pie.
Possibility of issue with Drea renthal?
 
![screenshot_20190214-121851](https://user-images.githubusercontent.com/27784255/52811835-564fcd00-3053-11e9-82ed-c841e1fa32b6.jpg)

**Comment by rune-b** [comment]
I get the same bug. You are correct, Drea's ability is the source of the ability

Stack trace

MissingReferenceException: The object of type 'GameObject' has been destroyed but you are still trying to access it.
Your script should either check if it is null or you should not destroy the object.
Ship.GenericShipBase.GetGlobalPoint (Vector3 localPoint) (at Assets/Scripts/Model/Content/Core/Ship/ShipBases/GenericShipBase.cs:211)
Ship.GenericShipBase.GetGlobalPoints (System.Collections.Generic.List`1 localPoints) (at Assets/Scripts/Model/Content/Core/Ship/ShipBases/GenericShipBase.cs:204)
BoardTools.ShotInfoArc.CheckRange () (at Assets/Scripts/Model/Board/Info/ShotInfoArc.cs:29)
BoardTools.ShotInfoArc..ctor (Ship.GenericShip ship1, Ship.GenericShip ship2, Arcs.GenericArc arc) (at Assets/Scripts/Model/Board/Info/ShotInfoArc.cs:24)
BoardTools.ShotInfo.CheckRange () (at Assets/Scripts/Model/Board/Info/ShotInfo.cs:97)
BoardTools.ShotInfo..ctor (Ship.GenericShip ship1, Ship.GenericShip ship2, System.Collections.Generic.List`1 weapons) (at Assets/Scripts/Model/Board/Info/ShotInfo.cs:79)
BoardTools.Board.IsShipInArc (Ship.GenericShip source, Ship.GenericShip target) (at Assets/Scripts/View/Board/Board.cs:297)
Abilities.SecondEdition.DreaRenthalAbility+DiceModificationAction.IsDiceModificationAvailable () (at Assets/Scripts/Model/Content/SecondEdition/Pilots/BTLA4YWing/DreaRenthal.cs:61)
Ship.GenericShip.CanUseDiceModification (ActionsList.GenericAction action) (at Assets/Scripts/Model/Content/Core/Ship/GenericShipActions.cs:434)
Ship.GenericShip.AddAvailableDiceModification (ActionsList.GenericAction action) (at Assets/Scripts/Model/Content/Core/Ship/GenericShipActions.cs:418)
Abilities.SecondEdition.DreaRenthalAbility.AddDreaRenthalAbility (Ship.GenericShip ship) (at Assets/Scripts/Model/Content/SecondEdition/Pilots/BTLA4YWing/DreaRenthal.cs:47)
Ship.GenericShip.GenerateDiceModificationsNormal () (at Assets/Scripts/Model/Content/Core/Ship/GenericShipActions.cs:384)
Ship.GenericShip.GenerateDiceModifications (DiceModificationTimingType type) (at Assets/Scripts/Model/Content/Core/Ship/GenericShipActions.cs:355)
Combat.ShowDiceModificationButtons (DiceModificationTimingType type, Boolean isForced) (at Assets/Scripts/View/Combat/Combat.cs:44)
Players.GenericPlayer.UseDiceModifications (DiceModificationTimingType type) (at Assets/Scripts/Model/Players/GenericPlayer.cs:107)
Combat.SwitchToRegularDiceModifications () (at Assets/Scripts/View/Combat/Combat.cs:108)
Combat.ShowDiceModificationButtons (DiceModificationTimingType type, Boolean isForced) (at Assets/Scripts/View/Combat/Combat.cs:71)
Players.GenericPlayer.UseDiceModifications (DiceModificationTimingType type) (at Assets/Scripts/Model/Players/GenericPlayer.cs:107)
Combat.SwitchToAfterRolledDiceModifications () (at Assets/Scripts/View/Combat/Combat.cs:122)
Players.GenericAiPlayer.UseDiceModifications (DiceModificationTimingType type) (at Assets/Scripts/Model/Players/GenericAiPlayer.cs:288)
SubPhases.DiceRollCombatSubPhase.CheckResults (.DiceRoll diceRoll) (at Assets/Scripts/Model/Phases/SubPhases/Temporary/DiceRollCombatSubPhase.cs:97)
SubPhases.AttackDiceRollCombatSubPhase.CheckResults (.DiceRoll diceRoll) (at Assets/Scripts/Model/Combat/Combat.cs:602)
SubPhases.DiceRollCombatSubPhase.FinallyCheckResults (.DiceRoll diceroll) (at Assets/Scripts/Model/Phases/SubPhases/Temporary/DiceRollCombatSubPhase.cs:84)
SubPhases.DiceRollCombatSubPhase.<ImmediatelyAfterRolling>m__1 () (at Assets/Scripts/Model/Phases/SubPhases/Temporary/DiceRollCombatSubPhase.cs:79)
Triggers.DoCallBack () (at Assets/Scripts/Model/Tools/Triggers.cs:314)
Triggers.ResolveTriggers (TriggerTypes triggerType, System.Action callBack) (at Assets/Scripts/Model/Tools/Triggers.cs:243)
Ship.GenericShip.CallOnImmediatelyAfterRolling (.DiceRoll diceroll, System.Action callBack) (at Assets/Scripts/Model/Content/Core/Ship/GenericShipCombat.cs:319)
SubPhases.DiceRollCombatSubPhase.ImmediatelyAfterRolling () (at Assets/Scripts/Model/Phases/SubPhases/Temporary/DiceRollCombatSubPhase.cs:79)
SubPhases.DiceRollCombatSubPhase.<RollDice>m__0 (.DiceRoll ) (at Assets/Scripts/Model/Phases/SubPhases/Temporary/DiceRollCombatSubPhase.cs:53)
DiceRoll.ExecuteCallback () (at Assets/Scripts/Model/Board/Dice/DiceRoll.cs:639)
ReplaysManager.ExecuteWithDelay (System.Action callback, Int32 seconds) (at Assets/Scripts/Model/Replays/ReplaysManager.cs:83)
DiceRoll.SyncDiceResults (System.Collections.Generic.List`1 sides) (at Assets/Scripts/Model/Board/Dice/DiceRoll.cs:634)
GameCommands.SyncDiceResultsCommand.Execute () (at Assets/Scripts/Model/GameController/GameCommands/SyncDiceResultsCommand.cs:26)
GameCommands.GameCommand.TryExecute () (at Assets/Scripts/Model/GameController/GameCommands/GameCommand.cs:107)
GameController.SendCommand (GameCommands.GameCommand command) (at Assets/Scripts/Model/GameController/GameController.cs:40)
GameModes.LocalGame.ExecuteServerCommand (GameCommands.GameCommand command) (at Assets/Scripts/Model/GameMode/LocalGame.cs:23)
Players.HumanPlayer.SyncDiceResults () (at Assets/Scripts/Model/Players/HumanPlayer.cs:145)
DiceRoll.CalculateWaitedResults () (at Assets/Scripts/Model/Board/Dice/DiceRoll.cs:607)
DiceRoll.CheckDiceMovementFinish () (at Assets/Scripts/Model/Board/Dice/DiceRoll.cs:576)
ShipMovementScript.UpdateSubscribedFuncs () (at Assets/Scripts/Model/Movement/ShipMovementScript.cs:50)
ShipMovementScript.Update () (at Assets/Scripts/Model/Movement/ShipMovementScript.cs:31)

**Comment by Sandrem** [comment]
Finally fixed.

**Comment by XPav** [comment]
So I just had a crash using Drea.   I destroyed Drea, and then the Lok fired, and I got the console when I think it tried to re-roll.   (First time I got the console, so I had to go get the editor log because I closed it).


NullReferenceException: Object reference not set to an instance of an object
  at Abilities.SecondEdition.DreaRenthalAbility+DiceModificationAction.IsDiceModificationAvailable () [0x00000] in <filename unknown>:0 
  at Ship.GenericShip.CanUseDiceModification (ActionsList.GenericAction action) [0x00000] in <filename unknown>:0 
  at Ship.GenericShip.AddAvailableDiceModification (ActionsList.GenericAction action) [0x00000] in <filename unknown>:0 
  at Abilities.SecondEdition.DreaRenthalAbility.AddDreaRenthalAbility (Ship.GenericShip ship) [0x00000] in <filename unknown>:0 
  at (wrapper delegate-invoke) Ship.GenericShip/EventHandlerShip:invoke_void__this___GenericShip (Ship.GenericShip)
  at (wrapper delegate-invoke) Ship.GenericShip/EventHandlerShip:invoke_void__this___GenericShip (Ship.GenericShip)
  at Ship.GenericShip.GenerateDiceModificationsNormal () [0x00000] in <filename unknown>:0 
  at Ship.GenericShip.GenerateDiceModifications (DiceModificationTimingType type) [0x00000] in <filename unknown>:0 
  at Combat.ShowDiceModificationButtons (DiceModificationTimingType type, Boolean isForced) [0x00000] in <filename unknown>:0 
  at Players.GenericPlayer.UseDiceModifications (DiceModificationTimingType type) [0x00000] in <filename unknown>:0 
  at Combat.SwitchToRegularDiceModifications () [0x00000] in <filename unknown>:0 
  at Combat.ShowDiceModificationButtons (DiceModificationTimingType type, Boolean isForced) [0x00000] in <filename unknown>:0 
  at Players.GenericPlayer.UseDiceModifications (DiceModificationTimingType type) [0x00000] in <filename unknown>:0 
  at Combat.SwitchToAfterRolledDiceModifications () [0x00000] in <filename unknown>:0 
  at Players.GenericAiPlayer.UseDiceModifications (DiceModificationTimingType type) [0x00000] in <filename unknown>:0 
  at SubPhases.DiceRollCombatSubPhase.CheckResults (.DiceRoll diceRoll) [0x00000] in <filename unknown>:0 
  at SubPhases.AttackDiceRollCombatSubPhase.CheckResults (.DiceRoll diceRoll) [0x00000] in <filename unknown>:0 
  at SubPhases.DiceRollCombatSubPhase.FinallyCheckResults (.DiceRoll diceroll) [0x00000] in <filename unknown>:0 
  at SubPhases.DiceRollCombatSubPhase.<ImmediatelyAfterRolling>m__1 () [0x00000] in <filename unknown>:0 
  at Triggers.DoCallBack () [0x00000] in <filename unknown>:0 
  at Triggers.ResolveTriggers (TriggerTypes triggerType, System.Action callBack) [0x00000] in <filename unknown>:0 
  at Ship.GenericShip.CallOnImmediatelyAfterRolling (.DiceRoll diceroll, System.Action callBack) [0x00000] in <filename unknown>:0 
  at SubPhases.DiceRollCombatSubPhase.ImmediatelyAfterRolling () [0x00000] in <filename unknown>:0 
  at SubPhases.DiceRollCombatSubPhase.<RollDice>m__0 (.DiceRoll ) [0x00000] in <filename unknown>:0 
  at DiceRoll.ExecuteCallback () [0x00000] in <filename unknown>:0 
  at ReplaysManager.ExecuteWithDelay (System.Action callback, Int32 seconds) [0x00000] in <filename unknown>:0 
  at DiceRoll.SyncDiceResults (System.Collections.Generic.List`1 sides) [0x00000] in <filename unknown>:0 
  at GameCommands.SyncDiceResultsCommand.Execute () [0x00000] in <filename unknown>:0 
  at GameCommands.GameCommand.TryExecute () [0x00000] in <filename unknown>:0 
  at GameController.SendCommand (GameCommands.GameCommand command) [0x00000] in <filename unknown>:0 
  at GameModes.LocalGame.ExecuteServerCommand (GameCommands.GameCommand command) [0x00000] in <filename unknown>:0 
  at Players.HumanPlayer.SyncDiceResults () [0x00000] in <filename unknown>:0 
  at DiceRoll.CalculateWaitedResults () [0x00000] in <filename unknown>:0 
  at DiceRoll.CheckDiceMovementFinish () [0x00000] in <filename unknown>:0 
  at ShipMovementScript.UpdateSubscribedFuncs () [0x00000] in <filename unknown>:0 
  at ShipMovementScript.Update () [0x00000] in <filename unknown>:0
